### PR TITLE
Fix #3046 UI Add support for request description

### LINKED ISSUE #3046: UI: Add support for request description
Refer #2645 [mocks](https://github.com/open-metadata/OpenMetadata/issues/2645#issuecomment-1044144728)

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #3046 
### Describe your changes :

I worked on issue #3046.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@shahsank3t, @darth-coder00

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx [hunk 1]
```diff
@@ -865,15 +905,20 @@ const ServicePage: FunctionComponent = () => {
...
-    getFeedCount(getEntityFeedLink(serviceCategory.slice(0, -1), serviceFQN))
+  const createThread = (data: CreateThread) => {
+    postThread(data)
       .then((res: AxiosResponse) => {
-        setFeedCount(res.data.totalCount);
+        setEntityThread((pre) => [...pre, res.data]);
+        getEntityFeedCount();
+        showToast({
+          variant: 'success',
+          body: 'Thread is created successfully',
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:915]:
```suggestion
          body: 'Conversation created successfully',
```

File: openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx [hunk 2]
```diff
@@ -865,15 +905,20 @@ const ServicePage: FunctionComponent = () => {
...
+        });
       })
       .catch(() => {
         showToast({
           variant: 'error',
-          body: 'Error while fetching feed count',
+          body: 'Error while creating thread',
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:921]:
```suggestion
          body: 'Error while creating the conversation',
```

File: openmetadata-ui/src/main/resources/ui/src/utils/FeedElementUtils.tsx [hunk 3]
```diff
@@ -33,13 +35,28 @@ export const getFieldThreadElement = (
...
-      <i className="far fa-comment" /> {threadValue.count} threads
+      <SVGIcons alt="comments" icon={Icons.COMMENT} width="18px" />{' '}
+      {threadValue.count}
     </p>
   ) : null;
 };
+
+export const getDefaultValue = (owner: EntityReference) => {
+  if (isUndefined(owner)) {
+    return;
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/utils/FeedElementUtils.tsx:52]: If the owner is not defined, can we have the message without the mention? The users will then have an option to tag a user with the message.

**Review by vivekratnavel — CHANGES_REQUESTED** [review]
Overall looks good to me. I have added inline comments/suggestions to improve.

**Comment by Sachin-chaurasiya** [comment]
@vivekratnavel, Thank you for the review. I'll look into it.

File: openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityFeedEditor/ActivityFeedEditor.tsx [hunk 4]
```diff
@@ -71,15 +74,22 @@ const ActivityFeedEditor: FC<ActivityFeedEditorProp> = ({
           trigger="mouseenter">
           <Button
             className={classNames(
-              'tw-bg-gray-400 tw-py-0.5 tw-px-1 tw-rounded',
+              'tw-py-0.5 tw-px-1 tw-rounde tw-bg-none',
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityFeedEditor/ActivityFeedEditor.tsx:77]: tw-rounded?
```suggestion
              'tw-py-0.5 tw-px-1 tw-rounded tw-bg-none',
```

**Comment by vivekratnavel** [comment]
- This placeholder looks good. One minor improvement could be to include the actual icon in the place of "chat icon" text.

Screen Shot 2022-03-03 at 10 36 21 AM
Something like this
Screen Shot 2022-03-03 at 10 42 03 AM

- The "Start new conversation" icon renders with different sizes and the position of (+) is not consistent.
Screen Shot 2022-03-03 at 10 34 45 AM

- Also, the start conversation icon should only be shown on hover just like the edit icon. The tags column should also show this icon on hover to start a new conversation around tags.
- The request description text here should be a button to improve UX. Here it matches with the same color as the disabled "No description added" text and doesn't look like an action item that the user can click on. Also, in my opinion, the text can be "No description"
Screen Shot 2022-03-03 at 10 46 53 AM
It can even be as subtle as this not to grab attention but be present everywhere. The button can brighten up on hover with a different border color (primary)
Screen Shot 2022-03-03 at 11 00 25 AM

File: openmetadata-ui/src/main/resources/ui/src/utils/FeedUtils.ts [hunk 5]
```diff
@@ -78,11 +79,15 @@ export const HTMLToMarkdown = new TurndownService({
     },
   });
 
-export const getReplyText = (count: number) => {
+export const getReplyText = (
+  count: number,
+  singular?: string,
+  plural?: string
+) => {
   if (count === 0) return 'Reply in thread';
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/utils/FeedUtils.ts:87]:
```suggestion
  if (count === 0) return 'Reply in conversation';
```

File: openmetadata-ui/src/main/resources/ui/src/utils/SvgUtils.tsx [hunk 6]
```diff
@@ -18,13 +18,15 @@ import IconGithub from '../assets/img/icon-github.png';
 import IconGoogle from '../assets/img/icon-google.png';
 import IconOkta from '../assets/img/icon-okta.png';
 import IconWelcomePopper from '../assets/img/welcome-popper-icon.png';
+import IconCommentPlus from '../assets/svg/add-chat01.svg';
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/utils/SvgUtils.tsx:21]: Use better SVG names instead of adding numbers

**Review by darth-coder00 — CHANGES_REQUESTED** [review]
One minor comment, otherwise LGTM!

**Review by darth-coder00 — APPROVED** [review]
LGTM!

**Comment by vivekratnavel** [comment]
Selenium tests are failing and some of them might be related to the changes in this patch:
```
2022-03-04T19:11:30.4443127Z [ERROR] Failures: 
2022-03-04T19:11:30.4454436Z [ERROR]   CommonTests.addMultipleTagsCheck:196 expected [11] but found [0]
2022-03-04T19:11:30.4454848Z [ERROR]   CommonTests.searchNotShowingResultsCheck:409 No search results found
2022-03-04T19:11:30.4456594Z [ERROR]   DashboardDetailsPageTest.editChartDescription:180 Description not updated
2022-03-04T19:11:30.4458326Z [ERROR]   DashboardServiceTestPage.checkConnectionConfigTab:137 Error while updating service
2022-03-04T19:11:30.4460066Z [ERROR]   DatabaseServicePageTest.checkConnectionConfigTab:159 Error while updating service
2022-03-04T19:11:30.4461845Z [ERROR]   MyDataPageTest.checkRecentlyViewed:274 expected [dim_address] but found [shopify]
2022-03-04T19:11:30.4462772Z [ERROR]   PipelineDetailsPageTest.editTaskDescription:187 Description not updated
2022-03-04T19:11:30.4463292Z [ERROR]   RolesPageTest.addRole:90 Role not added
2022-03-04T19:11:30.4464027Z [ERROR]   RolesPageTest.addRuleWithoutOperation:222->addRole:90 Role not added
2022-03-04T19:11:30.4464530Z [ERROR]   TagsPageTest.addTagWithExistingName:231 expected [Not Used] but found [1]
2022-03-04T19:11:30.4465066Z [ERROR]   TeamsPageTest.addAsset:166->createTeam:102 Team not added
2022-03-04T19:11:30.4465546Z [ERROR]   TeamsPageTest.addUser:112->createTeam:102 Team not added
2022-03-04T19:11:30.4466085Z [ERROR]   TeamsPageTest.checkTeamsFilterCount:214->createTeam:102 Team not added
2022-03-04T19:11:30.4466525Z [ERROR]   TeamsPageTest.createTeam:102 Team not added
2022-03-04T19:11:30.4467234Z [ERROR]   TeamsPageTest.editDescription:148->createTeam:102 Team not added
2022-03-04T19:11:30.4483251Z [ERROR]   TeamsPageTest.ownerNameIsConsistentCheck:195->addAsset:166->createTeam:102 Team not added
2022-03-04T19:11:30.4484889Z [ERROR] Errors: 
2022-03-04T19:11:30.4497903Z [ERROR]   CommonTests.overviewLinksAfterTour:317 » Timeout Expected condition failed: wa...
2022-03-04T19:11:30.4498745Z [ERROR]   CommonTests.tourStepSkippingCheck:329 » Timeout Expected condition failed: wai...
2022-03-04T19:11:30.4499483Z [ERROR]   DatabaseServicePageTest.checkIngestionTab:141 » Timeout Expected condition fai...
2022-03-04T19:11:30.4500133Z [ERROR]   MyDataPageTest.checkFollowingTab:260 » NoSuchElement no such element: Unable t...
2022-03-04T19:11:30.4500758Z [ERROR]   RolesPageTest.checkDuplicateRoleName:206 » Timeout Expected condition failed: ...
2022-03-04T19:11:30.4501355Z [ERROR]   RolesPageTest.editDescription:107 » Timeout Expected condition failed: waiting...
2022-03-04T19:11:30.4501991Z [ERROR]   TableDetailsPageTest.checkBreadCrumb:275 » NoSuchElement no such element: Unab...
2022-03-04T19:11:30.4502653Z [ERROR]   TableDetailsPageTest.checkFrequentlyJoinedColumns:320 » Timeout Expected condi...
2022-03-04T19:11:30.4505288Z [ERROR]   TableDetailsPageTest.checkFrequentlyJoinedTables:309 » Timeout Expected condit...
2022-03-04T19:11:30.4506187Z [ERROR]   TagsPageTest.TagUsageCheck:239 » Timeout Expected condition failed: waiting fo...
2022-03-04T19:11:30.4506929Z [ERROR]   TagsPageTest.removeTagWithExistingName:254 » Timeout Expected condition failed...
2022-03-04T19:11:30.4507601Z [ERROR]   TeamsPageTest.teamsWithSameDisplayNameCheck:251 » NoSuchElement no such elemen...
2022-03-04T19:11:30.4508280Z [ERROR]   TopicDetailsPageTest.checkFollow:112 » NoSuchElement no such element: Unable t...
2022-03-04T19:11:30.4508934Z [ERROR]   TopicDetailsPageTest.checkManage:198 » ElementClickIntercepted element click i...
2022-03-04T19:11:30.4509669Z [INFO] 
2022-03-04T19:11:30.4510096Z [ERROR] Tests run: 161, Failures: 16, Errors: 14, Skipped: 0
```

**Comment by vivekratnavel** [comment]
I have opened https://github.com/open-metadata/OpenMetadata/issues/3158 to fix Selenium test failures

**Review by vivekratnavel — APPROVED** [review]

**Pull request merged**